sklearn/multioutput.py

```diff
@@ -0,0 +1,208 @@
+    This allows for multiple target variable classifications. For each
+    target variable (column in y), a separate estimator is fit.
+
+    Parameters
+    ----------
+    estimator : estimator object
+        An estimator object implementing `fit` and `predict_proba`.
+
+    n_jobs : int, optional, default: 1
+        The number of jobs to use for the computation. If -1 all CPUs are used.
+        If 1 is given, no parallel computing code is used at all, which is
+        useful for debugging. For n_jobs below -1, (n_cpus + 1 + n_jobs) are
+        used. Thus for n_jobs = -2, all CPUs but one are used.
+        The number of jobs to use for the computation.
+
+        Note that parallel processing only occurs if there are multiple classes
+        within each target variable.
+        It does each target variable in y in series.
+
+    Attributes
+    __________
+    estimator: Sklearn estimator:
+        The base estimator used to construct the model.
+
+    """
+
+    def __init__(self, estimator=None, n_jobs=1):
+
+        self.estimator = estimator
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y):
+        """ Fit the model to data.
+        Creates a seperate model for each Response column.
+
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+            Data.
+
+        y : (sparse) array-like, shape = [n_samples, p_targets]
+            Multi-output targets. An indicator matrix turns on multilabel
+            classification.
+
+        Returns
+        -------
+        self
+        """
+
+        # check to see that the data is numeric
+
+        # check to see that the X and y have the same number of rows.
+        check_consistent_length(X, y)
+
+        # Calculate the number of classifiers
+        self._num_outputs = y.shape[1]
+
+        self.estimators_ = {}
+
+        for i in range(self._num_outputs):
+            estimator = clone(self.estimator)   # make a fresh clone
+            self.estimators_[i] = estimator.fit(X, y[:, i])
+
+        return self
+
+    def predict(self, X):
+        """Predict multi-class multiple target variable using a model
+         trained for each target variable.
+
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+            Data.
+
+        Returns
+        -------
+        y : (sparse) array-like, shape = [n_samples, p_targets]
+            Multi-output targets. An indicator matrix turns on multilabel
+            classification.
+            Predicted multi-class targets across multiple predictors.
+            Note:  entirely separate models are generated
+            for each predictor.
+        """
+        check_is_fitted(self, 'estimators_')
+
+        results = []
+        for i in range(self._num_outputs):
+            estimator = self.estimators_[i]
+            results.append(estimator.predict(X))
```

Comment:
I have a trivial doubt regarding this. If my array ``a`` is of shape ``(150, )``, then ``np.hstack((a,a))`` is giving ``(300, )`` while ``np.column_stack((a,a))`` is giving shape as ``(150, 2)``. Which is to be used here ?